crates/proposer-service/src/lib.rs

```diff
@@ -35,17 +41,39 @@ impl<L1Rpc> ProposerService<L1Rpc, proposer_client::Client<ProposerRpcClient, Ne
     pub fn new(config: &ProposerServiceConfig, l1_rpc: Arc<L1Rpc>) -> Result<Self, Error> {
         let proposer_rpc_client = ProposerRpcClient::new(config.client.proposer_endpoint.as_str())?;
         let network_prover = new_network_prover(config.client.sp1_cluster_endpoint.as_str());
+        let aggspan_vkey_hash = config.aggregated_proof_vkey_hash;
+
         Ok(Self {
             l1_rpc,
             client: Arc::new(proposer_client::Client::new(
                 proposer_rpc_client,
                 network_prover,
                 Some(config.client.proving_timeout),
             )?),
+            aggspan_vkey_hash,
         })
     }
 }
 
+fn check_aggspan_proof(
+    sp1_proof: &SP1ProofWithPublicValues,
+    expected_vkey_hash: B256,
+) -> Result<(), Error> {
+    let sp1_proof = &sp1_proof.proof;
+    let proof = &**sp1_proof
+        .try_as_compressed_ref()
+        .ok_or_else(|| Error::UnsupportedAggProofMode(sp1_proof.into()))?;
+
+    let vk = &proof.vk;
+    let vkey_hash = B256::from(vk.hash_bytes());
```

Comment:
Maybe using [hash_u32()](https://github.com/succinctlabs/sp1/blob/dev/crates/prover/src/types.rs#L47-L48) in all the places where we hash the aggregation vkey so that we could use the same functions to also check that it is properly equal to the `AGGREGATION_VKEY_HASH` hardcoded in `aggchain-proof-core` [here](https://github.com/agglayer/provers/blob/ff47b25d255bc0738154c316bf6383a71c06d7c1/crates/aggchain-proof-core/src/full_execution_proof.rs#L11-L13)

Reply:
Ok. will switch to `hash_u32`.

Regarding the vkey value, I introduced a config value (with a default) for this. Does it make sense for it to be a config value or should it be just a constant hardcoded in code? If it's in config, it would have to be passed into the proof. It probably does not make sense to have both. Thoughts? @atanmarko @Freyskeyd